kafka/src/main/java/com/linecorp/armeria/server/logging/structured/kafka/KafkaStructuredLogWriter.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2016 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.logging.structured.kafka;
+
+import java.util.concurrent.Future;
+
+import javax.annotation.Nullable;
+
+import org.apache.kafka.clients.producer.Callback;
+import org.apache.kafka.clients.producer.Producer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.linecorp.armeria.common.RequestContext;
+import com.linecorp.armeria.common.logging.ResponseLog;
+import com.linecorp.armeria.server.logging.structured.StructuredLogWriter;
+
+/**
+ * Kafka backend for service log logging.
+ * This class enable Kafka as a service log logging backend.
+ * As a typical sue, you can give constructed Kafka {@link Producer} and the topic which the service logs
+ * will be stored.
+ * {@code
+ * Producer<byte[], ApacheThriftStructuredLog> producer = new KafkaProducer<>(...);
+ * StructuredLogWriter<ApacheThriftStructuredLog> writer =
```

Comment:
ApacheThriftRpcStructuredLog? or We could rename ApacheThriftRpcStructuredLog to ApacheThriftStructuredLog given that Thrift implies Rpc.